src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -110,8 +108,23 @@ public void fromJson(String json) {
     }
   }
 
+  // Truncate working map to only print first 500 characters
   @Override
   public String toString() {
-    return new ToStringBuilder(this, ToStringStyle.JSON_STYLE).append("map", map).toString();
+    int truncateLength = 500;
+    StringBuilder sb = new StringBuilder("{");
+    map.forEach(
+        (k, v) -> {
+          int valLength = v.toString().length();
+          sb.append(
+              k
+                  + "="
+                  + v.toString()
+                      .substring(0, valLength > truncateLength ? truncateLength : valLength)
+                  + ", ");
+        });
+    sb.delete(sb.length() - 2, sb.length());
```

Comment:
It would be nice to not have to do it this way. Also, what if `map` is empty? I think you'd get an exception.

IMO using a stream over the map would be better. Something like
```
map.entrySet().stream().map(...).collect(Collectors.joining(", "))
```

Reply:
So much better. Thank you!